skippy-core/src/main/java/io/skippy/core/TestImpactAnalysis.java

```diff
@@ -134,48 +134,62 @@ PredictionWithReason predict(String testClassName, SkippyConfiguration configura
             if (NOT_FOUND.equals(this)) {
                 return PredictionWithReason.execute(new Reason(TEST_IMPACT_ANALYSIS_NOT_FOUND, Optional.empty()));
             }
-            var maybeAnalyzedTest = analyzedTests.stream()
-                    .filter(test -> classFileContainer.getById(test.getTestClassId()).getClassName().equals(testClassName))
-                    .findFirst();
-            if (maybeAnalyzedTest.isEmpty()) {
+
+            var testClassIds = getTestClassIds(testClassName);
+            var maybeAnalyzedTests = analyzedTests.stream()
+                    .filter(test -> testClassIds.contains(test.getTestClassId()))
+                    .toList();
```

Comment:
Even when working in a mono-module project, we can no longer try to match a single class ID to a single class name, since going forward a class name can have more than one ID (the ID assigned to the top-level class, plus the ID assigned to every nested class contained therein, if any).